netcore/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project(netcore LANGUAGES NONE)
+
+add_custom_target(netcore ALL COMMAND dotnet build WORKING_DIRECTORY = ${CMAKE_CURRENT_SOURCE_DIR}/NetCore DEPENDS apache-geode VERBATIM)
```

Comment:
I don't think that equals sign is supposed to be there. I am pretty surprised it doesn't scream at you or treat the path as literally "= /path/to/source/NetCore".

Reply:
Doesn't this technically depend on the C-bindings target?

Reply:
Actually no, it's a runtime dependency.  PInvoke code just says "look for this function in apache-geode-c.DLL", build doesn't verify such a file exists or anything.

Reply:
>Just a runtime dependency.  PInvoke code just says "look for this function in apache-geode-c.DLL", build doesn't verify such a file exists or anything.  We could make it dependent in cmake, but I don't think we'll really save anyone much time or trouble in the process.  The entire .net core build is < 5 seconds on my laptop.

Reply:
Ha! I'm just as surprised as you are.

Reply:
> >Just a runtime dependency.  PInvoke code just says "look for this function in apache-geode-c.DLL", build doesn't verify such a file exists or anything.  We could make it dependent in cmake, but I don't think we'll really save anyone much time or trouble in the process.  The entire .net core build is < 5 seconds on my laptop.
> 
> 

Ok, same difference. Runtime or compile time dependency, the dependency should be on the c binding target. If I ask CMake to build dotnet I would expect it to also build the runtime dependencies for it if they haven't or have changed. In this case any c binding changes would be missed.

Reply:
This should also probably pass the current configuration to the dotnet so that it can build either `Debug` or `DebWithRelInfo` rather than just the default in the csproj.

```
add_custom_target(netcore ALL 
  COMMAND COMAND dotnet build --configuration $<CONFIG>
  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/NetCore
  DEPENDS apache-geode
  VERBATIM)
```
See https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

Reply:
Still disagree on the dependencies - IMO CMakeLists.txt dependencies should enforce build-time dependencies _only_, and this ain't that. Either netcore or c bindings failing will still fail the entire build, but adding the dependency complicates the build graph and introduces potential to slow things down or introduce bugs.

Reply:
Completely agree on passing on configuration, tho.

Reply:
> Still disagree on the dependencies - IMO CMakeLists.txt dependencies should enforce build-time dependencies _only_, and this ain't that. Either netcore or c bindings failing will still fail the entire build, but adding the dependency complicates the build graph and introduces potential to slow things down or introduce bugs.

Not entirely true. Make a change to C-binding to correct or change behavior, change .NET Core library to compensate for changes, now when you compile the .NET Core tests to validate your new behavior it fails at runtime because you also didn't build the C-binding. You have to go build the C-binding manually to pick up that change then run the test. While not strictly a compile time dependency it is a very hard runtime dependency. Even runtime dependencies should be checked to make sure we have the correct version and since this one is source based the correct version may require a compilation.